components/tidb_query_datatype/src/codec/table.rs

```diff
@@ -32,6 +32,12 @@ pub const MAX_OLD_ENCODED_VALUE_LEN: usize = 9;
 
 /// Flag that indicate if the index value has common handle.
 pub const INDEX_VALUE_COMMON_HANDLE_FLAG: u8 = 127;
+/// Flag that indicate if the index value has partition id.
+pub const INDEX_VALUE_PARTITION_ID_FLAG: u8 = 126;
+/// Flag that indicate if the index value has restored data.
+pub const INDEX_VALUE_RESTORED_DATA_FLAG: u8 = 128;
```

Comment:
I prefer to use this constant to define this one: `pub const INDEX_VALUE_RESTORED_DATA_FLAG: u8 = CODEC_VERSION`